packages/plugins/typescript/operations/tests/ts-documents.spec.ts

```diff
@@ -7423,5 +7423,108 @@ function test(q: GetEntityBrandDataQuery): void {
         export type UserFragmentFragment = { __typename?: 'User', id: string } & { ' $fragmentName'?: 'UserFragmentFragment' };
       `);
     });
+
+    it("'mask' with @unmask yields correct types", async () => {
+      const ast = parse(/* GraphQL */ `
+        query {
+          me {
+            ...UserFragment @unmask
+          }
+        }
+        fragment UserFragment on User {
+          id
+        }
+      `);
+      const result = await plugin(
+        schema,
+        [{ location: 'test-file.ts', document: ast }],
+        { inlineFragmentTypes: 'mask' },
+        { outputFile: '' }
+      );
+      expect(result.content).toBeSimilarStringTo(`
+        export type Unnamed_1_QueryVariables = Exact<{ [key: string]: never; }>;
+
+
+        export type Unnamed_1_Query = { __typename?: 'Query', me?: (
+            { __typename?: 'User', id: string }
+            & { ' $fragmentRefs'?: { 'UserFragmentFragment': UserFragmentFragment } }
+          ) | null };
+
+        export type UserFragmentFragment = { __typename?: 'User', id: string } & { ' $fragmentName'?: 'UserFragmentFragment' };
+      `);
+    });
+
+    it("'mask' with @unmask and masked fragments yields correct types", async () => {
+      const ast = parse(/* GraphQL */ `
+        query {
+          me {
+            ...UserFragment @unmask
+            ...UserFragment2
+          }
+        }
+        fragment UserFragment on User {
+          id
+        }
+
+        fragment UserFragment2 on User {
+          email
+        }
+      `);
+      const result = await plugin(
+        schema,
+        [{ location: 'test-file.ts', document: ast }],
+        { inlineFragmentTypes: 'mask' },
+        { outputFile: '' }
+      );
+      expect(result.content).toBeSimilarStringTo(`
+        export type Unnamed_1_QueryVariables = Exact<{ [key: string]: never; }>;
+
+
+        export type Unnamed_1_Query = { __typename?: 'Query', me?: (
+            { __typename?: 'User', id: string }
+            & { ' $fragmentRefs'?: { 'UserFragmentFragment': UserFragmentFragment;'UserFragment2Fragment': UserFragment2Fragment } }
+          ) | null };
+
+        export type UserFragmentFragment = { __typename?: 'User', id: string } & { ' $fragmentName'?: 'UserFragmentFragment' };
+        export type UserFragment2Fragment = { __typename?: 'User', email: string } & { ' $fragmentName'?: 'UserFragment2Fragment' };
+      `);
+    });
+
+    it("'mask' with @unmask and masked fragments on overlapping fields yields correct types", async () => {
+      const ast = parse(/* GraphQL */ `
+        query {
+          me {
+            ...UserFragment @unmask
```

Comment:
Its unclear to me whether adding this functionality should be an extra option or built into the core functionality. Please advise. I'd be happy to change this up if we'd like to put an option around this.

Reply:
It could go either way I think. I'm checking in with the team but this shouldn't be blocking (as we can do another PR before releasing).

However, my personal opinion here is we don't need an option for now. This directive is unique to Apollo Client, so only Apollo Client users would try to use it. Other users won't notice any difference.

Reply:
Hi @jerelmiller 

I've checked with the team, since the directive is custom, it could be better to add an option. We are thinking that all custom directives can be under an parent `customDirectives` config. For example:

```
customDirectives: {
  apolloUnmask: true,
  // This can be used for other custom directives e.g.
  // relaySemanticNonNull: true,
}
```

Reply:
Ok great! I feel better about that rather than forcing a behavior with our specific implementation of the directive, so I will go this route. Thanks for checking on this!

Reply:
Added the new config option in [a9aee49](https://github.com/dotansimha/graphql-code-generator/pull/10163/commits/a9aee4957f4b59e693f5da5f26247114601d3884)